src/main/scala/com/kinja/soy/SoyValue.scala

```diff
@@ -42,35 +42,35 @@ case class SoyList(value: Seq[SoyValue]) extends AnyVal with SoyValue {
   def build = new SoyListData(asJavaIterable(value.map(_.build)))
 }
 
-case class SoyMap(fields: Seq[(String, SoyValue)]) extends AnyVal with SoyValue {
+case class SoyMap(value: Map[String, SoyValue]) extends AnyVal with SoyValue {
 
-  def build = new SoyMapData(mapAsJavaMap(Map(fields.map { case (k, v) => (k, v.build) }: _*)))
+  def build = new SoyMapData(mapAsJavaMap(value.mapValues(_.build)))
```

Comment:
`mapValues` creates a view of a map that's evaluated on every access to the map. You shouldn't use it unless you have a very good reason to do so. Maybe it's not a problem here, but I don't know what `mapAsJavaMap` does, so it's probably safer to use `.map { case (k, v) => (k, v) }` instead.

More on this here: http://blog.bruchez.name/2013/02/mapmap-vs-mapmapvalues.html